CMakeLists.txt

```diff
@@ -4,6 +4,67 @@ project(mrover)
 ## Compile as C++17, supported in ROS Kinetic and newer
 add_compile_options(-std=c++17 -Wall -Wextra -Werror -pedantic)
 
+## If on raspberry pi (RPi 4 w/ 64-bit Buster)
+if (${DEVICE} MATCHES "raspi4")
+    message("Building for device: ${DEVICE}")
+
+    find_package(catkin REQUIRED COMPONENTS  
+        rospy
+        std_msgs
+        message_generation
+        dynamic_reconfigure
+    )
+    catkin_python_setup()
+
+    ## Generate messages in the 'msg' folder
+    add_message_files(
+        FILES
+        Course.msg
+        Waypoint.msg
+        Chassis.msg
+    )
+
+    ## Generate services in the 'srv' folder
+    add_service_files(
+        FILES
+        PublishCourse.srv
+    )
+
+    ## Generate added messages and services with any dependencies listed here
+    generate_messages(
+        DEPENDENCIES
+        std_msgs
+    )
+
+    ## Generate dynamic reconfigure parameters in the 'cfg' folder
+    generate_dynamic_reconfigure_options(
+        config/DetectorParams.cfg
+    )
+
+    catkin_package(
+        #  INCLUDE_DIRS include
+        #  LIBRARIES mrover
+        CATKIN_DEPENDS roscpp rospy std_msgs message_runtime
+        #  DEPENDS system_lib
+    )
+
+    install(DIRECTORY launch/
+        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/launch
+        )
+    
+    # Python unit tests
+    catkin_add_nosetests(test/navigation/drive.py)
```

Comment:
go ahead and drop tests that aren't relevant to pi

Reply:
bump

Reply:
Ok